src/main/java/net/snowflake/ingest/connection/RequestBuilder.java

```diff
@@ -498,25 +526,31 @@ private URI makeGetClientURI(UUID requestId, String pipe) throws URISyntaxExcept
   }
 
   /**
-   * generateFilesJSON - Given a list of files, make some json to represent it
+   * Given a list of files, and an optional clientInfo generate json string which later can be
+   * passed in request body of insertFiles API
    *
    * @param files the list of files we want to send
+   * @param clientInfo optional clientInfo which can be empty.
    * @return the string json blob
    */
-  private String generateFilesJSON(List<StagedFileWrapper> files) {
+  private String serializeInsertFilesRequest(
+      List<StagedFileWrapper> files, Optional<InsertFilesClientInfo> clientInfo) {
     // if the files argument is null, throw
     if (files == null) {
       LOGGER.info("Null files argument in RequestBuilder");
       throw new IllegalArgumentException();
```

Comment:
It seems we also throw IllegalArugmentException back customer in this api, maybe we should also add that in our java doc?

Reply:
Will do!